pkg/errors/errors.go

```diff
@@ -11,60 +11,111 @@ import (
 	"github.com/kyokomi/emoji"
 )
 
-type LintRuleError struct {
-	Text     string
-	ID       string
-	ObjectID string
-	Value    any
-	Module   string
+type lintRuleError struct {
+	ID          string
+	Module      string
+	ObjectID    string
+	ObjectValue any
+	Text        string
 }
 
-func (l *LintRuleError) EqualsTo(candidate *LintRuleError) bool {
-	return l.ID == candidate.ID && l.Text == candidate.Text && l.ObjectID == candidate.ObjectID
+func (l *lintRuleError) EqualsTo(candidate lintRuleError) bool { //nolint:gocritic // it's a simple method
+	return l.ID == candidate.ID &&
+		l.Text == candidate.Text &&
+		l.ObjectID == candidate.ObjectID &&
+		l.Module == candidate.Module
 }
 
-func NewLintRuleError(id, objectID, module string, value any, template string, a ...any) *LintRuleError {
-	return &LintRuleError{
-		ObjectID: objectID,
-		Value:    value,
-		Text:     fmt.Sprintf(template, a...),
-		ID:       strings.ToLower(id),
-		Module:   module,
+type errStorage []lintRuleError
+
+type LintRuleErrorsList struct {
+	storage errStorage
+
+	linterID string
+	moduleID string
+	objectID string
+}
+
+func NewLinterRuleList(linterID string, module ...string) *LintRuleErrorsList {
+	l := &LintRuleErrorsList{
+		storage:  errStorage{},
+		linterID: linterID,
+	}
+	if len(module) > 0 {
+		l.moduleID = module[0]
 	}
+
+	return l
 }
 
-type LintRuleErrorsList struct {
-	data []*LintRuleError
+func (l *LintRuleErrorsList) WithObjectID(objectID string) *LintRuleErrorsList {
+	l.objectID = objectID
+
+	return l
 }
 
-// Add adds new error to the list if it doesn't exist yet.
-// It first checks if error is empty (i.e. all its fields are empty strings)
-// and then checks if error with the same ID, ObjectId and Text already exists in the list.
-func (l *LintRuleErrorsList) Add(e *LintRuleError) {
-	if e == nil {
-		return
+func (l *LintRuleErrorsList) WithModuleID(moduleID string) *LintRuleErrorsList {
```

Comment:
Shouldn't it be the `WithModuleName` or just `WithModule`.
ModuleID argument is very confusing. We don't use it anywhere